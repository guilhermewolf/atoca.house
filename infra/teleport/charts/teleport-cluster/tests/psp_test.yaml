suite: PodSecurityPolicy
templates:
  - psp.yaml
tests:
  - it: creates a PodSecurityPolicy when enabled in values and supported
    capabilities:
      majorVersion: 1
      minorVersion: 22
    set:
      podSecurityPolicy:
        enabled: true
    asserts:
      - hasDocuments:
          count: 3
      - documentIndex: 0
        isKind:
          of: PodSecurityPolicy
      - documentIndex: 1
        isKind:
          of: Role
      - documentIndex: 2
        isKind:
          of: RoleBinding
      - matchSnapshot: {}

  - it: does not create a PodSecurityPolicy when enabled in values but not supported
    set:
      podSecurityPolicy:
        enabled: true
    capabilities:
      majorVersion: 1
      minorVersion: 25
    asserts:
      - hasDocuments:
          count: 0
