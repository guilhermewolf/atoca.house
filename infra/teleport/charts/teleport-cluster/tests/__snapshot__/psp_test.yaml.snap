creates a PodSecurityPolicy when enabled in values and supported:
  1: |
    apiVersion: policy/v1beta1
    kind: PodSecurityPolicy
    metadata:
      annotations:
        seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default,runtime/default
        seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default
      name: RELEASE-NAME
    spec:
      allowPrivilegeEscalation: false
      fsGroup:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      hostIPC: false
      hostNetwork: false
      hostPID: false
      privileged: false
      readOnlyRootFilesystem: true
      requiredDropCapabilities:
      - ALL
      runAsUser:
        rule: MustRunAsNonRoot
      seLinux:
        rule: RunAsAny
      supplementalGroups:
        ranges:
        - max: 65535
          min: 1
        rule: MustRunAs
      volumes:
      - '*'
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: RELEASE-NAME-psp
      namespace: NAMESPACE
    rules:
    - apiGroups:
      - policy
      resourceNames:
      - RELEASE-NAME
      resources:
      - podsecuritypolicies
      verbs:
      - use
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: RELEASE-NAME-psp
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: RELEASE-NAME-psp
    subjects:
    - kind: ServiceAccount
      name: RELEASE-NAME
